src/tlo/methods/tb.py

```diff
@@ -2499,6 +2404,46 @@ def apply(self, person_id, squeeze_factor):
                     )
 
 
+class HSI_Tb_EndOfLifeCare(HSI_Event, IndividualScopeEventMixin):
+    """
+    this is a hospital stay for terminally-ill patients with TB
+    it does not affect disability weight or probability of death
+    no consumables are logged but health system capacity (HR) is allocated
+    there are no consequences if hospital bed is not available as person has scheduled death
+    already within 2 weeks
+    """
+
+    def __init__(self, module, person_id, beddays):
+        super().__init__(module, person_id=person_id)
+        assert isinstance(module, Tb)
+
+        self.TREATMENT_ID = "Tb_PalliativeCare"
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({})
+        self.ACCEPTED_FACILITY_LEVEL = "2"
+
+        self.beddays = beddays
+        self.BEDDAYS_FOOTPRINT = (
+            self.make_beddays_footprint({"general_bed": self.beddays})
+            if self.beddays
+            else self.make_beddays_footprint({"general_bed": 7.5})
+        )
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+        hs = self.sim.modules["HealthSystem"]
+
+        if not df.at[person_id, "is_alive"]:
+            return hs.get_blank_appt_footprint()
+
+        if df.at[person_id, "hv_art"] == "virally_suppressed":
+            return hs.get_blank_appt_footprint()
```

Comment:
@tbhallett I'm not sure what you mean here, the blank_appt_footprint would still allow a bed to be requested? Or the person would in real life require a bed?

Reply:
It seemed surprising to me. 
This will mean that someone who is virally suppressed with still occupy a bed. (but not take up the time of having palliative care). Will they die if they are virally suppressed?

Would we ideally want the HSI not to request a bed at all if the person is virally suppressed?